Unprefix -webkit-hyphens/-webkit-hyphenate-character CSS properties

joonghunpark opened the pull request:
#### 9f181fb3edafbb35640668393bf540e3fe69093f
<pre>
Unprefix -webkit-hyphens/-webkit-hyphenate-character CSS properties
<a href="https://bugs.webkit.org/show_bug.cgi?id=243670">https://bugs.webkit.org/show_bug.cgi?id=243670</a>

Reviewed by Tim Nguyen.

This change supports unprefixed hyphenate-character, not only -webkit-hyphenate-character.

According to the csswg discussion, w3c/csswg-drafts#6887, hyphenate-character css property is stable enough to ship.

* LayoutTests/TestExpectations:
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/fast/css/getComputedStyle/resources/property-names.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-pseudo/parsing/marker-supported-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-pseudo/parsing/marker-supported-properties-in-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-003-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-004-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-004.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-005-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-005.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-limit-chars-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-limit-chars-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-003-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-and-contenteditable-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-control-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-control.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-character-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-character.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-010.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-011.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-012.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-013.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-inline-010.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-inline-011.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-inline-012.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-011.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-012-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-012.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-013-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-013.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-014-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-014.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-015-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-015.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-out-of-flow-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-out-of-flow-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-punctuation-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-punctuation-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-shaping-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-shaping-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-span-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-span-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-003-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-004-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-004.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/shy-styling-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphenate-character-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphenate-character-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphenate-limit-chars-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphens-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphens-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/hyphens-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001-expected.txt:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/wpe/TestExpectations:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle):
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
(HTMLConverter::computedAttributesForElement):
* Source/WebCore/style/PropertyAllowlist.cpp:
(WebCore::Style::isValidMarkerStyleProperty):

Canonical link: <a href="https://commits.webkit.org/264628@main">https://commits.webkit.org/264628@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8f0876902fd3e2245350d42a60f3a809d9ef4a4f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8162 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8453 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8670 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9825 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8235 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/8169 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10441 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/8364 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11106 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8307 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9374 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7410 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9947 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6674 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/7468 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15026 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7797 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7597 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10959 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8068 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6555 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7362 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/7380 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1969 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11569 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7810 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/animation/CSSPropertyAnimation.cpp:
```diff
@@ -3940,7 +3940,7 @@ CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap()
 #endif
         case CSSPropertyWebkitFontSizeDelta:
-        case CSSPropertyWebkitHyphenateCharacter:
+        case CSSPropertyHyphenateCharacter:
```
Separate bug, but we should support discrete animations on hyphenate-character

joonghunpark replied:
I see. I will address that with a separate bug.

nt1m commented on LayoutTests/TestExpectations:
```diff
@@ -2788,6 +2788,11 @@ webkit.org/b/214290 imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens
 webkit.org/b/214290 imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-shaping-001.html [ ImageOnlyFailure ]
 webkit.org/b/214290 imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-shaping-002.html [ ImageOnlyFailure ]
 webkit.org/b/214290 imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-span-002.html [ ImageOnlyFailure ]
```
Can we remove the prefix on those pre-existing tests? (re-importing web-platform-tests/css/css-text/hyphens with this patch should do it for you)

nt1m replied:
These 14 failures make me wonder if this is ready to be enabled. would be good to know how significant each of them are. Are you planning to look into them?

joonghunpark replied:
> Can we remove the prefix on those pre-existing tests? (re-importing web-platform-tests/css/css-text/hyphens with this patch should do it for you)

Ok, I will re-import the web-platform-tests/css/css-text/hyphens test cases.

joonghunpark replied:
> These 14 failures make me wonder if this is ready to be enabled. would be good to know how significant each of them are. Are you planning to look into them?

FYI, when https://chromium-review.googlesource.com/c/chromium/src/+/3812328 unprefixed this property,
it changed TestExpectations as below, which is the same remaning failure list as in this patch.

crbug.com/1140728 external/wpt/css/css-text/hyphens/hyphens-span-002.html [ Failure ]
crbug.com/1290460 [ Mac ] external/wpt/css/css-text/letter-spacing/letter-spacing-bengali-yaphala-001.html [ Failure ]
-crbug.com/1281208 external/wpt/css/css-text/hyphens/hyphenate-character-001.html [ Failure ]
 crbug.com/1281208 external/wpt/css/css-text/hyphens/hyphenate-character-002.html [ Failure ]
-crbug.com/1281208 external/wpt/css/css-text/hyphens/hyphenate-character-003.html [ Failure ]
-crbug.com/1281208 external/wpt/css/css-text/hyphens/hyphenate-character-004.html [ Failure ]
 crbug.com/1281208 external/wpt/css/css-text/hyphens/hyphenate-character-005.html [ Failure ]
 crbug.com/1281208 external/wpt/css/css-text/hyphens/hyphens-vertical-001.html [ Failure ]
 crbug.com/1281208 external/wpt/css/css-text/hyphens/hyphens-vertical-002.html [ Failure ]
 crbug.com/1281208 external/wpt/css/css-text/hyphens/hyphens-vertical-003.html [ Failure ]
-crbug.com/1281208 external/wpt/css/css-text/hyphens/hyphens-vertical-004.html [ Failure ]

So I think we could enable this unprefixed ones with the remaining failures,
but if you want to hold it, we could do that until more those failures are fixed.

I'm planning to fix those failures with a separate bugs/patches.

nt1m replied:
It would be nice to have the failures (especially the ones that were already marked before this patch) related to the `hyphens` CSS property to be at least at Chrome parity level.

joonghunpark replied:
I will check and work on it. Maybe need to see hyphen related both inline formatting context and css.
For now, with a little more rebaselining I will land this change.

nt1m reviewed (commented):
Looks good generally.

joonghunpark committed:
Unprefix -webkit-hyphens/-webkit-hyphenate-character CSS properties
https://bugs.webkit.org/show_bug.cgi?id=243670

Reviewed by Tim Nguyen.

This change supports unprefixed hyphenate-character, not only -webkit-hyphenate-character.

According to the csswg discussion, w3c/csswg-drafts#6887, hyphenate-character css property is stable enough to ship.

* LayoutTests/TestExpectations:
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/fast/css/getComputedStyle/resources/property-names.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-pseudo/parsing/marker-supported-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-pseudo/parsing/marker-supported-properties-in-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-003-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-004-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-004.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-005-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-character-005.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-limit-chars-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphenate-limit-chars-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-003-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-and-contenteditable-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-control-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-control.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-character-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-character.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-010.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-011.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-012.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-013.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-inline-010.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-inline-011.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-manual-inline-012.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-011.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-012-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-012.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-013-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-013.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-014-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-014.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-015-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-none-015.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-out-of-flow-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-out-of-flow-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-punctuation-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-punctuation-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-shaping-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-shaping-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-span-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-span-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-003-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-004-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-004.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/hyphens/shy-styling-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphenate-character-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphenate-character-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphenate-limit-chars-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphens-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/hyphens-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/hyphens-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001-expected.txt:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/wpe/TestExpectations:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle):
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
(HTMLConverter::computedAttributesForElement):
* Source/WebCore/style/PropertyAllowlist.cpp:
(WebCore::Style::isValidMarkerStyleProperty):

Canonical link: https://commits.webkit.org/264628@main
